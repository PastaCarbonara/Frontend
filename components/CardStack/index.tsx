import React, {useContext, useRef} from 'react'
import Swiper from 'react-native-deck-swiper'
import Card from "../Card";
import {Recipe} from "../../types";
import {SessionWebsocketContext} from "../../contexts/SessionContext";

export default function CardStack({recipes = []}: { recipes: Recipe[] }) {

    const swiper = useRef<Swiper<any>>(null);
    const {isReady, lastMessage, send} = useContext(SessionWebsocketContext);

    function onLike() {
        if (swiper.current) {
            swiper.current.swipeRight();
        }
    }

    function onDislike() {
        if (swiper.current) {
            swiper.current.swipeLeft();
        }
    }

    function onSwipeLeft(cardIndex) {
        swipe(false, cardIndex)
    }

    function onSwipeRight(cardIndex) {
        swipe(true, cardIndex)
    }

    function swipe(isLike: boolean, cardIndex: number) {
        if (isReady) {
            send(JSON.stringify({
                    "action": "REQUEST_RECIPE_LIKE",
                    "payload": {
                        "like": isLike,
                        "recipe_id": recipes[cardIndex].id
                    }
                }
            ))
        }
    }

    return (
        <Swiper
            cards={recipes || []}
            renderCard={(card, index) => (
                <Card
                    recipe={card}
                    key={index}
                    onLike={onLike}
                    onDislike={onDislike}
                />
            )}
            ref={swiper}
            stackSize={3}
            stackSeparation={-20}
            backgroundColor={'#fff'}
            onSwipedLeft={onSwipeLeft}
            onSwipedRight={onSwipeRight}
        />
    )
}
