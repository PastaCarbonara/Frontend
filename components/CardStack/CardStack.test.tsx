import React from 'react';
import CardStack from '../cardStack';
import MockedNavigator from '../../jest/mocks/mockedNavigator';
import { render } from 'react-native-testing-library';

describe('CardStack', () => {
    it('should render correctly', () => {
        const { toJSON } = render(<MockedNavigator component={CardStack} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
