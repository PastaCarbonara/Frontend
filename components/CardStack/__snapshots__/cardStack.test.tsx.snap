// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardStack should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="MockedScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            pointerEvents="auto"
            style={
              [
                {
                  "alignItems": "stretch",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "backgroundColor": "#fff",
                  "marginBottom": 0,
                  "marginTop": 0,
                },
                {},
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  {
                    "zIndex": -3,
                  },
                ]
              }
            />
            <View
              collapsable={false}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                {
                  "flex": 1,
                  "height": 1214,
                  "left": 20,
                  "opacity": 1,
                  "position": "absolute",
                  "top": 60,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": 0,
                    },
                    {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 710,
                  "zIndex": 1,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "flex-end",
                      "backgroundColor": "#fff",
                      "borderRadius": 13,
                      "elevation": 6,
                      "flexDirection": "column",
                      "height": "90%",
                      "justifyContent": "flex-start",
                      "shadowColor": "#000000",
                      "shadowOffset": {
                        "height": 4,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.3,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    {
                      "borderRadius": 13,
                      "flex": 1,
                      "justifyContent": "flex-end",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": undefined,
                      }
                    }
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "height": undefined,
                          "width": "100%",
                        },
                        {
                          "borderRadius": 13,
                          "flex": 1,
                          "justifyContent": "flex-end",
                          "resizeMode": "cover",
                          "width": "100%",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      {
                        "padding": 15,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#fff",
                          "fontSize": 48,
                          "fontWeight": "bold",
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "flex-start",
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "position": "relative",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                            "width": 55,
                          }
                        }
                      >
                        <Text />
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                          }
                        }
                      >
                        <Text>
                          Bekijk recept
                        </Text>
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                            "width": 55,
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                {
                  "flex": 1,
                  "height": 1214,
                  "left": 20,
                  "position": "absolute",
                  "top": 60,
                  "transform": [
                    {
                      "scale": 0.97,
                    },
                    {
                      "translateY": -20,
                    },
                  ],
                  "width": 710,
                  "zIndex": -1,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "flex-end",
                      "backgroundColor": "#fff",
                      "borderRadius": 13,
                      "elevation": 6,
                      "flexDirection": "column",
                      "height": "90%",
                      "justifyContent": "flex-start",
                      "shadowColor": "#000000",
                      "shadowOffset": {
                        "height": 4,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.3,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    {
                      "borderRadius": 13,
                      "flex": 1,
                      "justifyContent": "flex-end",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": undefined,
                      }
                    }
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "height": undefined,
                          "width": "100%",
                        },
                        {
                          "borderRadius": 13,
                          "flex": 1,
                          "justifyContent": "flex-end",
                          "resizeMode": "cover",
                          "width": "100%",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      {
                        "padding": 15,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#fff",
                          "fontSize": 48,
                          "fontWeight": "bold",
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "flex-start",
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "position": "relative",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                            "width": 55,
                          }
                        }
                      >
                        <Text />
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                          }
                        }
                      >
                        <Text>
                          Bekijk recept
                        </Text>
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                            "width": 55,
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                {
                  "flex": 1,
                  "height": 1214,
                  "left": 20,
                  "position": "absolute",
                  "top": 60,
                  "transform": [
                    {
                      "scale": 0.9400000000000001,
                    },
                    {
                      "translateY": -40,
                    },
                  ],
                  "width": 710,
                  "zIndex": -2,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "flex-end",
                      "backgroundColor": "#fff",
                      "borderRadius": 13,
                      "elevation": 6,
                      "flexDirection": "column",
                      "height": "90%",
                      "justifyContent": "flex-start",
                      "shadowColor": "#000000",
                      "shadowOffset": {
                        "height": 4,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.3,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    {
                      "borderRadius": 13,
                      "flex": 1,
                      "justifyContent": "flex-end",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    source={
                      {
                        "uri": undefined,
                      }
                    }
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "height": undefined,
                          "width": "100%",
                        },
                        {
                          "borderRadius": 13,
                          "flex": 1,
                          "justifyContent": "flex-end",
                          "resizeMode": "cover",
                          "width": "100%",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      {
                        "padding": 15,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#fff",
                          "fontSize": 48,
                          "fontWeight": "bold",
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "flex-start",
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "position": "relative",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                            "width": 55,
                          }
                        }
                      >
                        <Text />
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                          }
                        }
                      >
                        <Text>
                          Bekijk recept
                        </Text>
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "borderRadius": 35,
                            "elevation": 6,
                            "flexDirection": "row",
                            "height": 55,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 5,
                            "shadowColor": "#000000",
                            "shadowOffset": {
                              "height": 4,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.3,
                            "width": 55,
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
